src/dftbp/dftb/rangeseparated.F90

```diff
@@ -8,1627 +8,6137 @@
+
+    !> True, if the derivative at xx is desired, otherwise the function value it returned
+    logical, intent(in), optional :: tDerivative
+
+    !! True, if the derivative at xx is desired, otherwise the function value it returned
+    !! Default: .false.
+    logical :: tPrime
+
+    !! Value of the polynomial at xx (in general should satisfy rDamp <= x <= rCut)
+    real(dp) :: yy
+
+    !! Polynomial coefficients
+    real(dp) :: aa, bb, cc, dd, ee, ff
+
+    !! rCut - rDamp
+    real(dp) :: delta
+
+    if (present(tDerivative)) then
+      tPrime = tDerivative
+    else
+      tPrime = .false.
+    end if
+
+    ! 5th order polynomial definition
+    ! f(x) = ax^5 + bx^4 + cx^3 + dx^2 + ex + f
+    ! f'(x) = 5ax^4 + 4bx^3 + 3cx^2 + 2dx + e
+    ! f''(x) = 20ax^3 + 12bx^2 + 6cx + 2d
+
+    ! boundary conditions:
+    ! f(rCut) = 0, f'(rCut) = 0, f''(rCut) = 0
+    ! f(rDamp) = y0, f'(rDamp) = y0p, f''(rDamp) = y0pp
+
+    delta = rCut - rDamp
+
+    aa = -(delta**2 * y0pp + 6.0_dp * delta * y0p + 12.0_dp * y0) / (2.0_dp * delta**5)
+
+    bb = (rCut * (5.0_dp * delta**2 * y0pp + 30.0_dp * delta * y0p + 60.0_dp * y0)&
+        & - 2.0_dp * delta**3 * y0pp - 14.0_dp * delta**2 * y0p - 30.0_dp * delta * y0)&
+        & / (2.0_dp * delta**5)
+
+    cc = -(rCut * (-8.0_dp * delta**3 * y0pp - 56.0_dp * delta**2 * y0p - 120.0_dp * delta * y0)&
+        & + rCut**2 * (10.0_dp * delta**2 * y0pp + 60.0_dp * delta * y0p + 120.0_dp * y0)&
+        & + delta**4 * y0pp + 8.0_dp * delta**3 * y0p + 20.0_dp * delta**2 * y0)&
+        & / (2.0_dp * delta**5)
+
+    dd = (rCut * (3.0_dp * delta**4 * y0pp + 24.0_dp * delta**3 * y0p + 60.0_dp * delta**2 * y0)&
+        & + rCut**2 * (-12.0_dp * delta**3 * y0pp - 84.0_dp * delta**2 * y0p - 180.0_dp * delta&
+        & * y0) + rCut**3 * (10.0_dp * delta**2 * y0pp + 60.0_dp * delta * y0p + 120.0_dp * y0))&
+        & / (2.0_dp * delta**5)
+
+    ee = -(rCut**2 * (3.0_dp * delta**4 * y0pp + 24.0_dp * delta**3 * y0p + 60.0_dp * delta**2&
+        & * y0) + rCut**3*(-8.0_dp * delta**3 * y0pp - 56.0_dp * delta**2 * y0p - 120.0_dp * delta&
+        & * y0) + rCut**4 * (5.0_dp * delta**2 * y0pp + 30.0_dp * delta * y0p + 60.0_dp * y0))&
+        & / (2.0_dp * delta**5)
+
+    ff = (rCut**3 * (delta**4 * y0pp + 8.0_dp * delta**3 * y0p + 20.0_dp * delta**2 * y0)&
+        & + rCut**4 * (-2.0_dp * delta**3 * y0pp - 14.0_dp * delta**2 * y0p - 30.0_dp * delta * y0)&
+        & + rCut**5 * (delta**2 * y0pp + 6.0_dp * delta * y0p + 12.0_dp * y0)) / (2.0_dp * delta**5)
+
+    if (tPrime) then
+      yy = 5.0_dp * aa * xx**4 + 4.0_dp * bb * xx**3 + 3.0_dp * cc * xx**2 + 2.0_dp * dd * xx + ee
+    else
+      yy = aa * xx**5 + bb * xx**4 + cc * xx**3 + dd * xx**2 + ee * xx + ff
+    end if
+
+  end function poly5zero
+
+
+  !> Returns the numerical second derivative of long-range gamma, by means of a central finite
+  !! difference.
+  function getddLrNumericalGammaValue(this, iSp1, iSp2, dist, delta) result(ddGamma)
```

Comment:
Should differentiate this analytically at some point...

Reply:
yes